nixos/modules/system/boot/tmp.nix

```diff
@@ -17,10 +25,11 @@ with lib;
     };
 
     boot.tmpOnTmpfs = mkOption {
-      type = types.bool;
+      type = with types; either bool string;
       default = false;
       description = ''
          Whether to mount a tmpfs on <filename>/tmp</filename> during boot.
+         Setting this to `true` is equivalent to 50% of available memory. For details about valid strings, see `man mount`.
```

Comment:
The 50% you are promising is something that will easily break, if upstream changes something. Either document that you do what upstream does, or implement the 50% yourself by figuring out available physical memory, etc.